src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {
+  useState, useContext, useRef, useCallback, useEffect,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../TodoContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({ todo }) => {
+  const { todos, setTodos, setChecked } = useContext(TodosContext);
+  const [title, setTitle] = useState(todo.title);
+  const [editing, setEditing] = useState(false);
+  const [focus, setFocus] = useState(false);
+  const thisTodo = useRef<HTMLInputElement>(null);
```

Comment:
```suggestion
  const inputRef = useRef<HTMLInputElement>(null);
```
Todo sounds like a todo object, etc.